cmd/sonrd/didutil/existing_did.go

```
@@ -0,0 +1,169 @@
+package didutil
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/did/ssi"
+	"github.com/sonr-io/sonr/x/object/types"
+)
+
+func surveyExistingDid() error {
+	file := ""
+	prompt := &survey.Input{
+		Message: "Enter the path to your DID Json file:",
+		Suggest: func(toComplete string) []string {
+			files, _ := filepath.Glob(toComplete + "*")
+			return files
+		},
+	}
+	survey.AskOne(prompt, &file)
+
+	buf, err := ioutil.ReadFile(file)
+	if err != nil {
+		return err
+	}
+	doc := &did.Document{}
+	err = doc.UnmarshalJSON(buf)
+	if err != nil {
+		return err
+	}
+
+	opt := ""
+	// Present public key types
+	selectOpt := &survey.Select{
+		Message: "What action would you like to perform?",
+		Options: []string{
+			"Add Assertion Method",
+			"Add Invocation Method",
+			"Add Verification Method",
+			"Encrypt JWE",
+			"Decrypt JWE",
+		},
+		Default: "Add Assertion Method",
+	}
+
+	// perform the questions
+	survey.AskOne(selectOpt, &opt, nil)
+
+	// Handle opt
+	switch opt {
+	case "Add Assertion Method":
+		// Add Assertion Method
+		controller := fmt.Sprintf("%s#test-%d", doc.ID, len(doc.Controller))
+		didController, err := did.ParseDID(controller)
+		if err != nil {
+			return err
+		}
+		priv := secp256k1.GenPrivKey()
+		pub := priv.PubKey()
+		vm, _ := did.NewVerificationMethod(doc.ID, ssi.ECDSASECP256K1VerificationKey2019, *didController, pub)
```

Don't ignore error

`cobra.CheckErr(err)`